### PR TITLE
Liquibase 3.6.3

### PR DESCRIPTION
**Pull request by Tcharl**
-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [x] Tests are added where necessary
-   [x] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

Also indented ejs and reordered plugins declaration by alphabetic

### CODE REVIEW

File: generators/server/templates/pom.xml.ejs [hunk 1]
```diff
@@ -993,51 +991,70 @@
...
-                            <version>${spring-boot.version}</version>
-                        </dependency>
-                        <dependency>
-                            <groupId>javax.validation</groupId>
-                            <artifactId>validation-api</artifactId>
-                            <version>${validation-api.version}</version>
-                        </dependency>
-                    </dependencies>
-                </plugin>
+            <plugin>
```
Comment (pascalgrimaud) [inline generators/server/templates/pom.xml.ejs]: not sure about the indent here

Comment (Tcharl) [reply generators/server/templates/pom.xml.ejs]: Hawkeye :-). Thank you Pascal

**Comment by Tcharl** [comment]
I do not think that my modification broke the build: WDYT?

**Comment by pascalgrimaud** [comment]
The current build is broken because of JDK11 + Infinispan. There is an opened ticket for that

File: generators/server/templates/pom.xml.ejs [hunk 2]
```diff
@@ -1019,13 +1016,11 @@
                             <artifactId>javassist</artifactId>
                             <version>${javassist.version}</version>
                         </dependency>
-                        <%_ if (databaseType === 'sql') { _%>
```
Comment (murdos) [inline generators/server/templates/pom.xml.ejs]: We should not depend on liquibase for database not handled by liquibase, e.g. cassandra.
Why are you removing this condition?

File: generators/server/templates/pom.xml.ejs [hunk 3]
```diff
@@ -105,11 +105,15 @@
         https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies/${spring-boot.version} -->
         <liquibase.version>3.6.3</liquibase.version>
         <liquibase-hibernate5.version>3.6</liquibase-hibernate5.version>
+    <%_ if (devDatabaseType === 'h2Disk') { _%>
+        <h2.version>1.4.199</h2.version>
+    <%_ } _%>
+        <spring.version>5.1.5.RELEASE</spring.version>
```
Comment (murdos) [inline generators/server/templates/pom.xml.ejs]: Why is this necessary? Shouldn't the version matches the version of spring-boot?

File: generators/server/templates/pom.xml.ejs [hunk 4]
```diff
@@ -1036,6 +1031,28 @@
...
+                            <artifactId>h2</artifactId>
+                            <version>${h2.version}</version>
+                        </dependency>
+        <%_ } _%>
+                        <dependency>
+                            <groupId>org.springframework</groupId>
+                            <artifactId>spring-core</artifactId>
+                            <version>${spring.version}</version>
+                        </dependency>
+                        <dependency>
```
Comment (murdos) [inline generators/server/templates/pom.xml.ejs]: Do you really need to declare the 3 dependencies? spring-context depends on spring-beans and spring-core.

File: generators/server/templates/pom.xml.ejs (hunk 2 above)
Comment (murdos) [reply generators/server/templates/pom.xml.ejs]: Ok, understood

File: generators/server/templates/pom.xml.ejs (hunk 4 above)
Comment (Tcharl) [reply generators/server/templates/pom.xml.ejs]: Yes it is, otherwise I wouldn't have put them.
Double checked...

File: generators/server/templates/pom.xml.ejs (hunk 3 above)
Comment (Tcharl) [reply generators/server/templates/pom.xml.ejs]: It should, but unfortunately its mandatory due to liquibase-maven-pg dependency reference

**Comment by Tcharl** [comment]
Everythings' ready IMO

**Comment by DanielFran** [comment]
@Tcharl Can you fix de conflicts?

**Comment by mraible** [comment]
Bump @Tcharl. This PR has conflicts.

**Comment by Tcharl** [comment]
@DanielFran @mraible rebased :-)

**Comment by DanielFran** [comment]
Let's see if this can be merged since @jdubois did not yet created the v6 release

**Comment by jdubois** [comment]
I want to do the release ASAP, but there's quite a lot of changes here, so I'm not sure what to do

**Comment by jdubois** [comment]
It's a bit of a mess because there are also quite a lot of formatting changes, and I'm not even sure they are correct, so I just can't merge this now. Is that fixing some important bug?

**Comment by jdubois** [comment]
Oh and plugins have been reordered, which can cause any kind of issue as this change the classpath, as that's not possible to test easily (you basically need to run everything to have a look). So I just can't merge this for v6, I hope there's nothing important here.

**Comment by Tcharl** [comment]
Sorry, I had a duplicate on failsafe while rebasing: it's removed now.

Concerning the formatting, it's only on ejs (using good tabulation number so that it's easier to see the logic): no impact on the generated code at all (and it was useful, I noticed a duplicate conditional ;-)).

But as you said, nothing important: it's only a liquibase update and not top priority, this is just ready to ship.

**Comment by jdubois** [comment]
OK thanks @Tcharl - that's too complex for me if I want to release this evening, so it will be in the next release.

**Comment by Tcharl** [comment]
@jdubois , look like it is important: see #9646 . Looks like the liquibase version has already been bumped without testing...

**Comment by jdubois** [comment]
Thanks @Tcharl but as on #9646 the user doesn't guide is version number, there's no way for us to know... if he's a newcomer (which I expect as he doesn't follow the guidelines), then that would be strange that he uses a beta version

**Comment by jdubois** [comment]
OK @Tcharl after reviewing this doesn't look too risky, let's merge this for JHipster 6!

**Comment by Tcharl** [comment]
It's in the beta for sure: the master has already a 3.6.3 liquibase version on the pom, while the 5.8.2 had 3.5.5.

3.6.3 liquibase need some additional dependencies (the ones on that PR), but 3.5.5 does not.

Thanks for merging: otherwise maven users wouldn't have benefit from working liquibase with v6 :-)

**Comment by jdubois** [comment]
Great, thanks a lot for your help @Tcharl

**Comment by jdubois** [comment]
@Tcharl normally we don't put $100 tags on PRs but I added this one especially for you, as this definitely solves a big issue in the upcoming v6 release - don't hesitate to claim the bug bounty!

**Comment by Tcharl** [comment]
Thank you!

I'll pay my JHipster conference ticket with these bucks, and the according beers :-).

Is it possible to have a jhipster-conf Paris ticket instead :-p? Or should I ask for the bounty then buy the ticket (and I'll be happy to give 20$ from the 100$ I earned to the project at the same occasion :-p).

**Comment by jdubois** [comment]
@Tcharl why not have both? As a top contributor to the project (and also an Ippon client!) I can probably grab you some free tickets -> can you send me a message on my personal GMail account so I don't forget?

**Comment by Tcharl** [comment]
No, really, I don't want both: I'm already ashamed to have failed making my company being a sponsor of that project, and I don't want to get profit from the project until that goal will be reached.
I know, it's a stupid matter of honor, but I want it to be safe ;-)

**Comment by jdubois** [comment]
Don't worry for the sponsoring, we should be good this year - if you really want to help, I would be very happy to have more attendees, maybe your company could send more people? I could even get a group discount

**Comment by Tcharl** [comment]
I'll try to send the newer team working on the product I made, also, I'll ask if there are some interested people on our Digital branch ;-)
